### ISSUE TITLE
[BUG_SP] Incorrect detection of field boundaries for grape/olive fields at a 0 degree angle

### ISSUE THREAD

**Issue by JoneZ42** [Courseplay/Courseplay_FS22#3177]
**Steps to reproduce**
Place a Grape or Olive field in a exact 0-degree angle ( maybe same at 90° - was not tested by me)
Set up a field job

**Actual behavior**
The field boundary is narrow and runs diagonally to the corners, the course is a single diagonal line.

**Expected behavior**
The field boundary is detected and the course is set up correctly

**Screenshots**
![courseplay olives](https://github.com/Courseplay/Courseplay_FS22/assets/11922526/c6575e77-aaba-4960-a258-b0dbad5d7007)

**Courseplay version**
Courseplay for FS22 v7.4.1.1

**Log**
[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/14765157/log.txt)

**Map**
Erlengrad

**Vehicles/ implements involved**
Which vehicle or implement? - All Vehicles
Mod yes/no ? - No

**Relevant script mods**
Tested in a Savegame with only Curseplay on

**Comment by Tensuko** [comment]
Thats not the complete logfile.

**Comment by Tensuko** [comment]
Tried as perfect straight as possible. No problems.
![grafik](https://github.com/Courseplay/Courseplay_FS22/assets/27344744/c499e66b-73fd-4af9-a161-8f43b52a5d10)

**Comment by JoneZ42** [comment]
Hi,
Oh dear, I think I found a edge-case - I'm feeling like the most accurate farmer on earth...

Here is a "HowTo: I create a perfect 0-degree olive field." 

1. Make a Straight line with the painting tool or use a field border as orientation.
![fsScreen_2024_03_30_16_12_03](https://github.com/Courseplay/Courseplay_FS22/assets/11922526/873e2749-e990-4989-9a37-1f543d4d58ba)
2. Choose the olives and activate the "raster mode" (C) - in German, it's "Raster Platzierung"
3. While placing the first row, try to figure out if on or beside the orientation line, place it when you think it fits
![fsScreen_2024_03_30_16_15_28](https://github.com/Courseplay/Courseplay_FS22/assets/11922526/79f99ccd-e7b8-42e5-b3c6-3e3caaa32a22)
4. Control in TopDown View, if it's 100% accurate (no "steps" in the field border) - the longer the row is, the better it can be seen - to be sure
![fsScreen_2024_03_30_16_16_07](https://github.com/Courseplay/Courseplay_FS22/assets/11922526/43997297-305e-4e84-9667-d044a5145ec2)
5. expand the field as usual

Then, set up the course, etc.
![fsScreen_2024_03_30_16_17_35](https://github.com/Courseplay/Courseplay_FS22/assets/11922526/d8346289-670d-4323-a335-72ce8231569e)
![fsScreen_2024_03_30_16_18_05](https://github.com/Courseplay/Courseplay_FS22/assets/11922526/a89cd975-f256-4c26-8387-24399edcadc2)



Here is the log file of the complete process.
[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/14812508/log.txt)


It's a fresh savegame with only couseplay on and the one tractor as you can see in the screenshots, but I tried to create a course with other machines - it's the same (if the field is created as above)

**Comment by Tensuko** [comment]
Because you painted the field.
I also just used the the field border.

Do not paint it.
Use I think it is F4 do show the grid if you rly need it.

**Comment by JoneZ42** [comment]
I don't now, i think its not becouse of the painting...
![grafik](https://github.com/Courseplay/Courseplay_FS22/assets/11922526/65da8b2c-2f2b-46a9-954a-4db67b71e233)

btw unintended tested with 90° - it only seems to be the case at 0°

I don't want to bother you - i'm at a point where it feels like this, couse i can rly live with this bug

but still, thx for the advice with f4 - could be useful in other cases

**Comment by FarmerBoy2007** [comment]
Just ran into exactly the same issue, here's a log file from a fresh save game:
[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/14890455/log.txt)

An easy way I found to place your vines perfectly straight is to first place a single vine segment (as short as possible) with grid snapping on, and then extend it.

**Comment by Tensuko** [comment]
Why you actually have to do it 0.000° straight... Its a game guys.... To be honest if it is just happening with perfect straight by doing special tricks, I am tempted to close this as won't fix...

**Comment by JoneZ42** [comment]
You're right. It's just a game, but every person plays it differently. Some people like perfection.
I initially found this bug on a field I created when I played without course play. I wanted a field I could easily drive over with GPS, so I made one with - as it turned out later - 0°.

From my point of view, the bug does not need to be fixed. If I feel like it, I can create fields with 90° or 45° in perfect angles.

So long, and thx for your time
